dotCMS/src/main/java/com/dotmarketing/filters/CMSFilter.java

```diff
@@ -34,28 +34,34 @@
 
 public class CMSFilter implements Filter {
 
-    private final HttpServletRequestThreadLocal requestThreadLocal =
-            HttpServletRequestThreadLocal.INSTANCE;
+    private final HttpServletRequestThreadLocal requestThreadLocal = HttpServletRequestThreadLocal.INSTANCE;
     private CMSUrlUtil urlUtil = CMSUrlUtil.getInstance();
     private static VisitorAPI visitorAPI = APILocator.getVisitorAPI();
-    private final String RELATIVE_ASSET_PATH = APILocator.getFileAssetAPI()
-            .getRelativeAssetsRootPath();
+    private final String RELATIVE_ASSET_PATH = APILocator.getFileAssetAPI().getRelativeAssetsRootPath();
     public static final String CMS_INDEX_PAGE = Config.getStringProperty("CMS_INDEX_PAGE", "index");
 
     public enum IAm {
-        PAGE,
-        FOLDER,
-        FILE,
-        NOTHING_IN_THE_CMS
+        PAGE, FOLDER, FILE, NOTHING_IN_THE_CMS
     }
 
     @Override
     public void init(FilterConfig arg0) throws ServletException {
 
     }
 
-    public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
-            throws IOException, ServletException {
+
+    public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
+
+        try {
+            _doFilter(req, res, chain);
+        } finally {
+            DbConnectionFactory.closeSilently();
+        }
+
+
+    }
+
+    private void _doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 33 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This method has 87 lines, which is greater than the 75 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS138)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)